File: Python PowerPoint Examples/ShapesApiDemo/addSlideNewParagraph.py
import spirecloudpowerpoint
from spirecloudpowerpoint.configuration import *

appId = "your id"
appKey = "your key"
baseUrl = "https://api.e-iceblue.cn"
configuration = Configuration(appId, appKey,baseUrl)
api= spirecloudpowerpoint.api.shapes_api.ShapesApi(configuration)

fileName = "AddSlideNewParagraph.pptx"
password = ""
storage = ""
inputFolder = "input"

slide_index = 0
shape_index = "4"
source = spirecloudpowerpoint.models.Paragraph()
source.alignment = "Right"
source.text_alignment = "Top"
source.east_asian_line_break = False
source.indent = 2
source.margin_left = 1
source.margin_right = 5
source.space_after = 1
source.space_before = 5
result =api.add_slide_new_paragraph(fileName, slide_index, shape_index, source=source, password=password,storage=storage,folder=inputFolder)

File: JavaScript PowerPoint Examples/ShapesApiDemo/addSlideNewParagraph.js
(function (){
    var appId = "your ID";
    var appKey = "your Key";
    var baseUrl = "https://api.e-iceblue.cn";

    var SpirecloudPowerpoint = require('../../src/index');
    var apiClient = new SpirecloudPowerpoint.ApiClient(appId, appKey,baseUrl);
    var instance = new SpirecloudPowerpoint.ShapesApi(apiClient);
    
    var name = 'AddSlideNewParagraph.pptx';
    var folder = 'input';
    var slideIndex = 0;
    var shapeIndex = 0;
    var opts = {};
    opts.source = new SpirecloudPowerpoint.Paragraph();
    opts.source.MarginLeft = 1.0;
    opts.source.MarginRight = 5.0;
    opts.source.SpaceBefore = 5.0;
    opts.source.SpaceAfter = 1.0;
    opts.source.Indent = 2.0;
    opts.source.Alignment = "Right";
    opts.source.TextAlignment = "Top";
    opts.folder = folder;
    
    var response = instance.addSlideNewParagraph(name, slideIndex, shapeIndex, opts,
     function (error, data, response) {
         if (error) throw error;
     });
})();